Edit upload script

brynnz22 opened the pull request:
Added a line in the manifest upload script to account for Dataset Pubmed Id attribute as a string list. Had to convert to string first, then split.

Commit:
Add script to update legacy annotations and attribute/cv dictionary

Commit:
fix stringlist loop to account for integer lists

Commit:
fix stripping of 0 on grant numbers

Commit:
uncomment syn.store

vpchung commented on annotations/manifest_upload.py:
```diff
@@ -71,13 +71,14 @@ def edit_manifest(file_path, col_types_dict):
             # For columns with USERID as datatype, remove .0 tacked on in data type conversion.
-            if col_types_dict[columnName] == str:
+            if col_types_dict.get(columnName) == 'USERID':
                 df[columnName] = df[columnName].str.rstrip('.0')
```
Whoops!  I just learned that `rstrip()` reads the argument as a _list of characters_, **not** a substring.  That is:

```
s.rstrip(".0")
```

will remove `.` **or** `0` if they're found at the end of the string, NOT `.0` like I thought.  My bad!

For robustness, let's not use rstrip() at all (in case a user ID has a `0` at the end, which would then be stripped too).  Instead, let's use pandas' `replace()`:

```suggestion
                df[columnName] = df[columnName].replace("\.0$", "")
```

vpchung commented on annotations/manifest_upload.py:
```diff
@@ -71,13 +71,14 @@ def edit_manifest(file_path, col_types_dict):
+            df[columnName] = df[columnName].astype(str)
             df[columnName] = df[columnName].str.split(', ')
         else:
             for k, v in col_types_dict.items():
```
This loop is probably not necessary, since you're only looking at one column at a time. And since a column can only have one type, an iteration through the dict is a little superfluous.  You're also not using `k` or `v` at all, which is another hint that the loop is not needed!

So rather than this, I suggest looking up the type directly:

```suggestion
            df[columnName] = df[columnName].astype(
                col_types_dict[columnName])
```

vpchung replied:
Also, does `col_type_dict` contain all the of column types we need? If not, I suggest using `get()` so that we can set a default type, e.g. str, if a key is not found:

```
            df[columnName] = df[columnName].astype(
                col_types_dict.get(columnName, str))
```

brynnz22 replied:
Ah! Thank you for catching this!

vpchung reviewed (changes requested):
Some comments!

brynnz22 committed:
Update annotations/manifest_upload.py

delete .rstrip('.0') because it removes '.' OR '0' and we want to remove the entire string.

Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

brynnz22 committed:
Update annotations/manifest_upload.py

Delete unnecessary dictionary loop.

Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

vpchung reviewed (approved):
LGTM!

Do these changes still work as expected?

Commit:
fix indenting in for loop

brynnz22 commented:
Yep it works! I just had to fix the indentation in the for loop a little bit. I will merge it now. Thanks for the review!!!
